src/twisted/application/runner/_pidfile.py

```diff
@@ -0,0 +1,202 @@
+# -*- test-case-name: twisted.application.runner.test.test_pidfile -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+PID file.
+"""
+
+import errno
+from os import getpid, kill
+
+from twisted.python.filepath import IFilePath
+from twisted.logger import Logger
+
+
+
+class PIDFile(object):
+    """
+    PID file.
+
+    Manages a file that remembers a process ID.
+    """
+
+    _log = Logger()
+
+
+    @staticmethod
+    def format(pid=None):
```

Comment:
Yes… that was because… `write()` takes a `None` pid and probably at one point that made sense but there's no longer a need for that here.  Removing.